### PR TITLE
Webhook integration in ci

### PR DESCRIPTION
**Pull request by manish-agrawal-ai**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test case A
- [ ] Test case B

# Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas
* [ ] I have tested it for all user roles

### CODE REVIEW

File: api/restHandler/WebhookEventHandler.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	// validate signature
+	requestBodyBytes, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		impl.logger.Errorw("Cannot read the request body:", "err", err)
+		writeJsonResp(w, err, nil, http.StatusInternalServerError)
+		return
+	}
+	isValidSig := impl.webhookSecretValidator.ValidateSecret(r, secretFromRequest, requestBodyBytes, gitHost)
+	if !isValidSig {
+		impl.logger.Error("Signature mismatch")
```
Comment (nishant-d) [inline api/restHandler/WebhookEventHandler.go:83]: please add some meta info in log

File: internal/sql/repository/GitHostRepository.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	WebhookUrl  	string    `sql:"webhook_url"`
+	WebhookSecret   string    `sql:"webhook_secret"`
+	EventTypeHeader string    `sql:"event_type_header"`
+	SecretHeader    string    `sql:"secret_header"`
+	SecretValidator string    `sql:"secret_validator"`
+	models.AuditLog
+}
+
+type GitHostRepository interface {
+	FindAll() ([]GitHost, error)
```
Comment (nishant-d) [inline internal/sql/repository/GitHostRepository.go:39]: please use pointer to pass values in all methods

File: internal/sql/repository/pipelineConfig/CiWorkflowRepository.go [hunk 3]
```diff
@@ -87,8 +87,16 @@ type GitCommit struct {
 	Date    time.Time
 	Message string
 	Changes []string
+	WebhookData WebhookData
 }
 
+type WebhookData struct {
```
Comment (nishant-d) [inline internal/sql/repository/pipelineConfig/CiWorkflowRepository.go:93]: apply go code formatter

File: pkg/git/WebhookSecretValidator.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+const (
+	SECRET_VALIDATOR_SHA1 string = "SHA-1"
+	SECRET_VALIDATOR_URL_APPEND string = "URL_APPEND"
+	SECRET_VALIDATOR_PLAIN_TEXT string = "PLAIN_TEXT"
+)
+
+
+func (impl *WebhookSecretValidatorImpl) ValidateSecret(r *http.Request, secretInUrl string, requestBodyBytes []byte, gitHost *pipeline.GitHostRequest) bool {
```
Comment (nishant-d) [inline pkg/git/WebhookSecretValidator.go:51]: add some documentation around this block

**Review by nishant-d — COMMENTED** [review]
1. use pointer instead of objects
2. use go code formatting

File: api/restHandler/WebhookEventHandler.go (hunk 1 above)
Comment (manish-agrawal-ai) [reply api/restHandler/WebhookEventHandler.go:83]: Done

File: internal/sql/repository/GitHostRepository.go (hunk 2 above)
Comment (manish-agrawal-ai) [reply internal/sql/repository/GitHostRepository.go:39]: Taking it as my learning, will try to incorporate from next time onwards.

File: pkg/git/WebhookSecretValidator.go (hunk 4 above)
Comment (manish-agrawal-ai) [reply pkg/git/WebhookSecretValidator.go:51]: Done

File: internal/sql/repository/pipelineConfig/CiWorkflowRepository.go (hunk 3 above)
Comment (manish-agrawal-ai) [reply internal/sql/repository/pipelineConfig/CiWorkflowRepository.go:93]: Done

**Pull request closed without merging**